### PR TITLE
Prevent fatal error in set_current_gateway with no WC session

### PR DESCRIPTION
**Pull request by vedanshujain**
Follow up from #25242. Added unit tests since there weren't any for payment gateways.

### Changelog

> Fix - Fix fatal error that would occur when calling the WC_Payment_Gateways::set_current_gateway() function and there is no WC session.

### CODE REVIEW

File: tests/unit-tests/payment-gateways/payment-gateways.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	public function tearDown() {
+		parent::tearDown();
+		WC()->initialize_session();
+	}
+
+	/**
+	 * Test that we can set a current gateway from session. Basic smoke test.
+	 */
+	public function test_wc_set_current_gateway_from_session() {
+		WC()->initialize_session();\
```
Comment (hjoelr) [inline tests/unit-tests/payment-gateways/payment-gateways.php]: It looks like there is a typo in this line with a `\` at the end of the line.

**Review by hjoelr — APPROVED** [review]

**Comment by hjoelr** [comment]
Original issue is #25241

File: tests/unit-tests/payment-gateways/payment-gateways.php (hunk 1 above)
Comment (vedanshujain) [reply tests/unit-tests/payment-gateways/payment-gateways.php]: Nice catch. Fixed now.

**Review by rodrigoprimo — APPROVED** [review]
Looks good to me.

**Pull request merged**